gem2rpm/foreman_plugin.spec.erb

```diff
@@ -0,0 +1,184 @@
+<%
+has_assets = spec.files.any? { |f| f.start_with?('app/assets/') }
+has_apidoc = spec.files.any? { |f| f =~ %r{\Aapp/controllers/?.*/api/} }
+has_db_migrations = spec.files.any? { |f| f.start_with?('db/migrate/') }
+has_db_seeds = spec.files.any? { |f| f.start_with?('db/seeds.d/') }
+
+precompile_flags = ''
+precompile_flags << ' -a' if has_apidoc
+precompile_flags << ' -s' if has_assets
+
+config.rules[:doc] << /\/?CHANGES/
+config.rules[:ignore] << /.tx(\/.*)?/
+-%>
+
+Name: %{?scl_prefix}rubygem-%{gem_name}
+Version: <%= spec.version %>
+Release: 1%{?dist}
+Summary: <%= spec.summary.gsub(/\.$/, "") %>
+Group: Applications/Systems
+License: <%= spec.licenses.join(" and ") %>
+<% if spec.homepage -%>
+URL: <%= spec.homepage %>
+<% end -%>
+Source0: <%= download_path.sub('https:', 'http:') %>%{gem_name}-%{version}.gem
+Requires: foreman >= FIXME
+Requires: %{?scl_prefix_ruby}ruby(release)
+<% for req in spec.required_ruby_version -%>
+Requires: %{?scl_prefix_ruby}ruby <%= req %>
+<% end -%>
+<% for req in spec.required_rubygems_version -%>
+Requires: %{?scl_prefix_ruby}ruby(rubygems) <%= req %>
+<% end -%>
+<% for d in spec.runtime_dependencies -%>
+<% for req in d.requirement -%>
+Requires: %{?scl_prefix}rubygem(<%= d.name %>) <%= req  %>
+<% end -%>
+<% end -%>
+<% if has_assets -%>
+BuildRequires: foreman-assets
+<% end -%>
+BuildRequires: foreman-plugin >= FIXME
+<% if has_assets || has_apidoc -%>
+<% for d in spec.runtime_dependencies -%>
+<% for req in d.requirement -%>
+BuildRequires: %{?scl_prefix}rubygem(<%= d.name %>) <%= req  %>
+<% end -%>
+<% end -%>
+<% end -%>
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+<% for req in spec.required_ruby_version -%>
+BuildRequires: %{?scl_prefix_ruby}ruby<%= "-devel" unless spec.extensions.empty? %> <%= req %>
+<% end -%>
+<% for req in spec.required_rubygems_version -%>
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel <%= req %>
+<% end -%>
+<% if spec.extensions.empty? -%>
+BuildArch: noarch
+<% end -%>
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+Provides: foreman-plugin-%{plugin_name}
+
+%description
+<%= spec.description %>
+
+<% if doc_subpackage -%>
+%package doc
+Summary: Documentation for %{pkg_name}
+Group: Documentation
+Requires: %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+BuildArch: noarch
+
+%description doc
+Documentation for %{pkg_name}.
+<% end # if doc_subpackage -%>
+
+%prep
+%{?scl:scl enable %{scl} - << \EOF}
+gem unpack %{SOURCE0}
+%{?scl:EOF}
+
+%setup -q -D -T -n  %{gem_name}-%{version}
+
+%{?scl:scl enable %{scl} - << \EOF}
+gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
+%{?scl:EOF}
+
+%build
+# Create the gem as gem install only works on a gem file
+%{?scl:scl enable %{scl} - << \EOF}
+gem build %{gem_name}.gemspec
+%{?scl:EOF}
+
+# %%gem_install compiles any C extensions and installs the gem into ./%%gem_dir
+# by default, so that we can move it into the buildroot in %%install
+%{?scl:scl enable %{scl} - << \EOF}
```

Comment:
It should work, in that the `scl enable` command will run both commands, but if the first command failed then the RPM build wouldn't fail so it's better to have them run separately. (The RPM scriptlets will run with set -e so any failed command causes a failure, but scl enable won't fail if one command fails.)